### PR TITLE
[GraphBolt][CUDA] RGCN example

### PR DESCRIPTION
**Pull request by mfbalin**
## Description
Adding GPU sampling support for the R-GCN example.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Comment by frozenbugs** [comment]
LGTM

File: examples/sampling/graphbolt/rgcn/hetero_rgcn.py [hunk 1]
```diff
@@ -657,7 +658,7 @@ def main(args):
     )
     parser.add_argument("--num_epochs", type=int, default=3)
     parser.add_argument("--num_workers", type=int, default=0)
-    parser.add_argument("--num_gpus", type=int, default=0)
+    parser.add_argument("--num_gpus", type=int, default=1)
```
Comment (Rhett-Ying) [inline examples/sampling/graphbolt/rgcn/hetero_rgcn.py:661]: default to 0?

File: examples/sampling/graphbolt/rgcn/hetero_rgcn.py [hunk 2]
```diff
@@ -133,17 +138,6 @@ def create_dataloader(
         node_feature_keys["institution"] = ["feat"]
     datapipe = datapipe.fetch_feature(features, node_feature_keys)
 
-    # Move the mini-batch to the appropriate device.
```
Comment (Rhett-Ying) [inline examples/sampling/graphbolt/rgcn/hetero_rgcn.py:136]: so in the latest example, we only support
1. cpu sampling+ cpu training
2. gpu sampling + gpu training
The hack here is for cpu sampling + gpu training. If so, it's ok to remove the hack.

**Review by Rhett-Ying — APPROVED** [review]

File: examples/sampling/graphbolt/rgcn/hetero_rgcn.py (hunk 1 above)
Comment (mfbalin) [reply examples/sampling/graphbolt/rgcn/hetero_rgcn.py:661]: Default to 1 because if there is no GPU available, it will be set back to 0 automatically.

Comment (mfbalin) [reply examples/sampling/graphbolt/rgcn/hetero_rgcn.py:661]: Why do we need this parameter? We could simply remove it?

File: examples/sampling/graphbolt/rgcn/hetero_rgcn.py (hunk 2 above)
Comment (mfbalin) [reply examples/sampling/graphbolt/rgcn/hetero_rgcn.py:136]: Yeah, that seems to be the case.

**Pull request merged**